chat/src/main/java/org/navgurukul/chat/core/repo/AuthenticationRepository.kt

```diff
@@ -0,0 +1,60 @@
+package org.navgurukul.chat.core.repo
+
+import android.content.Context
+import im.vector.matrix.android.api.MatrixCallback
+import im.vector.matrix.android.api.auth.AuthenticationService
+import im.vector.matrix.android.api.auth.data.HomeServerConnectionConfig
+import im.vector.matrix.android.api.auth.data.LoginFlowResult
+import im.vector.matrix.android.api.session.Session
+import org.navgurukul.chat.core.extensions.configureAndStart
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class AuthenticationRepository(
+    private val authService: AuthenticationService,
+    private val activeSessionHolder: ActiveSessionHolder,
+    private val appContext: Context
+) {
+
+    suspend fun getLoginFlow(): LoginFlowResult? = suspendCoroutine {
+        authService.getLoginFlow(
+            HomeServerConnectionConfig.Builder()
+                .withHomeServerUri(HOME_SERVER_URL)
+                .build(),
+            object : MatrixCallback<LoginFlowResult> {
+                override fun onSuccess(data: LoginFlowResult) {
+                    it.resume(data)
+                }
+
+                override fun onFailure(failure: Throwable) {
+                    it.resume(null)
+                }
+            }
+        )
+    }
+
+    suspend fun login(): Session? {
+        getLoginFlow()?.let { _ ->
+            return suspendCoroutine {
+                authService.getLoginWizard().login(
+                    "t-saral",
+                    "hello123",
+                    "Android",
```

Comment:
This hardcoding is expected? Shouldn't we read it from somewhere and won't vary from user to user